src/main/asciidoc/index.adoc

```diff
@@ -17,503 +17,801 @@
+====
+It's strongly RECOMMENDED to rely on a cryptographic signature for JWS integrity protection.
+====
+
+Conversely, the use of a Message Authentication Code (MAC) for integrity protection is strongly discouraged (as it would
+allow any compromised service that can validate a JWT to create a new JWT using the same key).
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Claims within a JWE are signed then encrypted, and as such are entirely opaque to clients.
+As such, they MAY hold sensitive information.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+As far as algorithms requirements for JWS and JWE go, it's strongly RECOMMEND to use only ES256, ES256K, or RS256.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The unsecured "none" algorithm MUST NOT be used.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+To select the verification algorithm, RESTful services MUST rely on their local configuration rather than on the
+information in the JWT header.
+====
+
+In addition to the access token, the OAuth 2.0 specification foresees another token type: the "refresh token".
+
+The refresh token is a long-lived - usually opaque - token that can be issued by the authorization server to a client to
+allow it to obtain a new access token when the current becomes invalid or expires.
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Just like the access token, the refresh token MUST be used at most once and MUST be kept confidential in transit and at
+rest.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+However, unlike the access token, a refresh token is intended to be used solely with authorization servers and MUST NOT
+be sent to resource servers.
+====
+
+The subset of capabilities delegated by a token can be specified through a scoping mechanism.
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The authorization server MUST allow the client to specify the scope of the access request using the "scope" request
+parameter.
+====
 
-==== How to create a Detached JWS
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+Since user consent is REQUIRED for public clients, the scope (resources and permissions) that the user is expected to
+grant MUST be explained in a clear and understandable manner.
+====
 
-image::image9.png[image,width=468,height=160]
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The authorization server MUST use the "scope" response parameter to inform the client of the scope of the issued access
+token.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+The scope of an access token MUST be as narrow as possible to adhere to the principle of least privilege.
+====
+
+[.rule, caption="Rule {counter:rule-number}: "]
+====
+As tempting as it may be to try to centralize all authorization logic on the authorization server, OAuth 2.0 scopes MUST
+NOT be used to represent permissions (i.e., user privileges).
```

Comment:
Probably requires further elaboration and discussion. I've also encountered use of scopes to imply a permission is granted, use of custom claims to indicate user roles specific to called API.

Reply:
In my opinion, this is a misuse of scopes (although you are right that it is not uncommon to see it).

Reply:
This will be discussed during the meeting with SMALS.